hip-service/Link/Patient/LinkPatient.cs

```diff
@@ -0,0 +1,137 @@
+                            "Unable to store data to Database")));
+                }
+
+                var expiry = new DateTime(DateTime.Today.Year, DateTime.Today.Month, DateTime.Today.Day,
+                        DateTime.Now.Hour, DateTime.Now.Minute + 1, DateTime.Now.Second).ToUniversalTime()
+                    .ToString(Constants.DateTimeFormat);
+
+                var meta = new LinkReferenceMeta(nameof(CommunicationMode.MOBILE),
+                    patient.PhoneNumber, expiry);
+                var patientLinkReferenceResponse = new PatientLinkReferenceResponse(new LinkReference(linkRefNumber,
+                    "MEDIATED", meta));
+
+                return new Tuple<PatientLinkReferenceResponse, ErrorResponse>(patientLinkReferenceResponse, null);
+            }).ValueOr(
+                Task.FromResult(new Tuple<PatientLinkReferenceResponse, ErrorResponse>(null
+                    , new ErrorResponse(new Error(ErrorCode.NoPatientFound,"No patient Found"))))
+            );
+        }
+
+        public async Task<Tuple<PatientLinkResponse, ErrorResponse>> VerifyAndLinkCareContext(HipLibrary.Patient.Models.Request.PatientLinkRequest request)
+        {
+            if (await patientVerification.Verify(request.LinkReferenceNumber, request.Token) != null)
+            {
+                return new Tuple<PatientLinkResponse, ErrorResponse>
+                    (null, new ErrorResponse(new  Error(ErrorCode.OtpInValid, "Otp Invalid")));   
+            }
+
+            var (linkRequest, exception) =
+                await linkPatientRepository.GetPatientReferenceNumber(request.LinkReferenceNumber);
+            
+            if (exception != null)
+            {
+                return new Tuple<PatientLinkResponse, ErrorResponse>
+                    (null, new ErrorResponse(new Error(ErrorCode.NoPatientFound, "No request found")));
+            }
+            
+            var patientInfo = patientRepository.GetPatientInfoWithReferenceNumber(linkRequest.PatientReferenceNumber);
+
+            return patientInfo.Map(patient =>
+            {
+                var representations = linkRequest.CareContexts
+                    .Where(careContext =>
+                        patient.CareContexts.Any(info => info.ReferenceNumber == careContext.CareContextNumber))
+                    .Select(context => new CareContextRepresentation(context.CareContextNumber,
+                        patient.CareContexts.First(info => info.ReferenceNumber == context.CareContextNumber)
+                            .Description));
+
+                var patientLinkResponse = new PatientLinkResponse(new HipLibrary.Patient.Models.Response.LinkPatient(
+                    linkRequest.PatientReferenceNumber
+                    , patient.FirstName + " " + patient.LastName, representations));
+                
+                return new Tuple<PatientLinkResponse, ErrorResponse>(patientLinkResponse, null);;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4769144)